packages/teams-js/src/public/dialog.ts

```diff
@@ -103,12 +110,30 @@ export namespace dialog {
    */
   export function submit(result?: string | object, appIds?: string | string[]): void {
     ensureInitialized(FrameContexts.content, FrameContexts.sidePanel, FrameContexts.task, FrameContexts.meetingStage);
+
     if (!isSupported()) {
       throw errorNotSupportedOnPlatform;
     }
 
-    // Send tasks.completeTask instead of tasks.submitTask message for backward compatibility with Mobile clients
-    sendMessageToParent('tasks.completeTask', [result, appIds ? (Array.isArray(appIds) ? appIds : [appIds]) : []]);
+    app
+      .getContext()
+      .then((appContext) => {
+        // if actionInfo exists in appContext, the dialog was loaded from a 3rd party app,
+        // and if submit was called with a result, throw an error because the result cannot be
+        // passed along to the desired recipient
+        if (result && appContext.actionInfo) {
```

Comment:
This code is making non-obvious assumptions that could change in the future. It's totally not obvious why actionInfo in the context would change the behavior of this API. I think it would be good to add a separate API here specific for these actions and a new FrameContext for this flow.

Reply:
that's why I put a code comment, after a conversation offline with @MarcinJuraszek and @ydogandjiev, we decided to talk about this when @TrevorJoelHarris is back.